subsys/net/lib/zperf/zperf_udp_uploader.c

```diff
@@ -160,6 +160,8 @@ void zperf_udp_upload(const struct shell *sh,
 			      PACKET_SIZE_MAX);
 		packet_size = PACKET_SIZE_MAX;
 	} else if (packet_size < sizeof(struct zperf_udp_datagram)) {
+		BUILD_ASSERT(sizeof(struct zperf_udp_datagram) <= PACKET_SIZE_MAX,
+			     "Invalid PACKET_SIZE_MAX");
```

Comment:
I'm just not sure if it's the right place for such a build assert? Wouldn't it be better to place it in `zperf_internal.h`, where `struct zperf_udp_datagram` and `PACKET_SIZE_MAX` are defined?

Reply:
Done

Unsure however about if and which header to include when using BUILD_ASSERT.

Reply:
The macro itself is defined in `zephyr/toolchain/common.h`, but this header is pulled by `__assert.h` so I guess it's fine.